controllers/perconaservermysql_controller.go

```diff
@@ -1531,3 +1539,179 @@ func (r *PerconaServerMySQLReconciler) restartGroupReplication(ctx context.Conte
+				OwnerReferences: ownerReferences,
+			},
+			Spec: cm.CertificateSpec{
+				SecretName: cr.Name + "-ca-cert",
+				CommonName: cr.Name + "-ca",
+				IsCA:       true,
+				IssuerRef: cmmeta.ObjectReference{
+					Name:  caIssuerName,
+					Kind:  issuerKind,
+					Group: issuerGroup,
+				},
+				Duration:    &metav1.Duration{Duration: time.Hour * 24 * 365},
+				RenewBefore: &metav1.Duration{Duration: 730 * time.Hour},
+			},
+		}
+
+		err = r.Create(context.TODO(), caCert)
+		if err != nil && !k8serrors.IsAlreadyExists(err) {
+			return errors.Wrap(err, "create CA certificate")
+		}
+
+		if err := r.waitForCerts(cr.Namespace, cr.Name+"-ca-cert"); err != nil {
+			return err
+		}
+
+		if err := r.createIssuer(ownerReferences, cr.Namespace, issuerName, caCert.Spec.SecretName); err != nil {
+			return err
+		}
+	}
+	hosts := []string{
+		fmt.Sprintf("*.%s-mysql", cr.Name),
+		fmt.Sprintf("*.%s-mysql.%s", cr.Name, cr.Namespace),
+		fmt.Sprintf("*.%s-mysql.%s.svc.cluster.local", cr.Name, cr.Namespace),
+		fmt.Sprintf("*.%s-orchestrator", cr.Name),
+		fmt.Sprintf("*.%s-orchestrator.%s", cr.Name, cr.Namespace),
+		fmt.Sprintf("*.%s-orchestrator.%s.svc.cluster.local", cr.Name, cr.Namespace),
+	}
+	kubeCert := &cm.Certificate{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:            cr.Name + "-ssl",
+			Namespace:       cr.Namespace,
+			OwnerReferences: ownerReferences,
+		},
+		Spec: cm.CertificateSpec{
+			SecretName: cr.Spec.SSLSecretName,
+			DNSNames:   hosts,
+			IsCA:       false,
+			IssuerRef: cmmeta.ObjectReference{
+				Name:  issuerName,
+				Kind:  issuerKind,
+				Group: issuerGroup,
+			},
+		},
+	}
+
+	if cr.Spec.TLS != nil && len(cr.Spec.MySQL.ServiceAccountName) > 0 {
+		kubeCert.Spec.DNSNames = append(kubeCert.Spec.DNSNames, cr.Spec.TLS.SANs...)
+	}
+
+	err = r.Create(context.TODO(), kubeCert)
+	if err != nil && !k8serrors.IsAlreadyExists(err) {
+		return errors.Wrap(err, "create certificate")
+	}
+
+	return r.waitForCerts(cr.Namespace, cr.Spec.SSLSecretName)
+}
+
+func (r *PerconaServerMySQLReconciler) waitForCerts(namespace string, secretsList ...string) error {
+	ticker := time.NewTicker(3 * time.Second)
+	timeoutTimer := time.NewTimer(30 * time.Second)
+	defer timeoutTimer.Stop()
+	defer ticker.Stop()
+	for {
+		select {
+		case <-timeoutTimer.C:
+			return errors.Errorf("timeout: can't get tls certificates from certmanager, %s", secretsList)
+		case <-ticker.C:
+			sucessCount := 0
+			for _, secretName := range secretsList {
+				secret := &corev1.Secret{}
+				err := r.Get(context.TODO(), types.NamespacedName{
+					Name:      secretName,
+					Namespace: namespace,
+				}, secret)
+				if err != nil && !k8serrors.IsNotFound(err) {
+					return err
+				} else if err == nil {
+					sucessCount++
+				}
+			}
+			if sucessCount == len(secretsList) {
+				return nil
+			}
+		}
+	}
+}
+
+func (r *PerconaServerMySQLReconciler) createIssuer(ownRef []metav1.OwnerReference, namespace, issuer string, caCertSecret string) error {
```

Comment:
it might be better to accept `cm.IssuerConfig` as an argument, rather than `caCertSecret`. that way reader can immediately understand what'll the function do